lib/motsu-proc/src/test.rs

```diff
@@ -15,37 +15,39 @@ pub(crate) fn test(_attr: &TokenStream, input: TokenStream) -> TokenStream {
     let fn_block = &item_fn.block;
     let fn_args = &sig.inputs;
 
-    // If the test function has no params, then it doesn't need access to the
-    // contract, so it is just a regular test.
-    if fn_args.is_empty() {
-        return quote! {
-            #( #attrs )*
-            #[test]
-            fn #fn_name() #fn_return_type {
-                let _lock = ::motsu::prelude::acquire_storage();
-                let res = #fn_block;
-                ::motsu::prelude::reset_storage();
-                res
-            }
-        }
-        .into();
+    // Currently, more than one contract per unit test is not supported.
+    if fn_args.len() > 1 {
```

Comment:
Nit: I would say we should require always 1 contract for simplicity -> `if fn_args.len() != 1`

Reply:
If we do not want to test a contract I do not see any reason for not using regular `#[test]`

Reply:
Currently it repeats the previous behavior. If no contract is declared as an argument, it will still clear storage associated with the current test thread. In case user declared contract by himself inside test, like: `Contract::default()`. Before this refactoring it would be a bug, but currently it just drops unnecessary recourse from `STORAGE`. Idk..

Reply:
Ok, let's leave as it is.